lib/node_modules/@stdlib/iter/while-each/test/test.js

```diff
@@ -0,0 +1,368 @@
+	t.end();
+});
+
+tape( 'the function throws an error if provided an iterator argument which is not an iterator protocol-compliant object', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		{},
+		[],
+		function noop() {}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			iterWhileEach( value, noop, noop );
+		};
+	}
+});
+
+tape( 'the function throws an error if provided a second argument which is not a function', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		[],
+		{}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			iterWhileEach( randu(), value, noop );
+		};
+	}
+});
+
+tape( 'the function throws an error if provided a third argument which is not a function', function test( t ) {
+	var values;
+	var i;
+
+	values = [
+		'5',
+		5,
+		NaN,
+		true,
+		false,
+		null,
+		void 0,
+		[],
+		{}
+	];
+
+	for ( i = 0; i < values.length; i++ ) {
+		t.throws( badValue( values[i] ), TypeError, 'throws an error when provided '+values[i] );
+	}
+	t.end();
+
+	function badValue( value ) {
+		return function badValue() {
+			iterWhileEach( randu(), noop, value );
+		};
+	}
+});
+
+tape( 'the function returns an iterator protocol-compliant object', function test( t ) {
+	var count;
+	var it;
+	var r;
+	var i;
+
+	it = iterWhileEach( randu(), predicate, assert );
+	t.equal( it.next.length, 0, 'has zero arity' );
+
+	count = 0;
+	i = 0;
+	do {
+		r = it.next();
+		if ( r.value ) {
+			t.equal( typeof r.value, 'number', 'returns a number' );
+		}
+		t.equal( typeof r.done, 'boolean', 'returns a boolean' );
+		if ( r.done ) {
+			count += 1;
+		}
+		i += 1;
+	} while ( r.done === false );
+	t.equal( count, i, 'returns expected value' );
+	t.end();
+
+	function assert( v, i ) {
+		count += 1;
+		t.equal( isnan( v ), false, 'is not NaN' );
+		t.equal( isnan( i ), false, 'is not NaN' );
+	}
+
+	function predicate( v, i ) {
+		return ( v <= 0.5 && i >= 0 );
+	}
+});
+
+tape( 'the function returns an iterator protocol-compliant object which invokes a function for each iterated value before returning the iterated value until either a `predicate` function returns `false` or the iterator has iterated over all values.', function test( t ) {
+	var expected;
+	var opts;
+	var rand;
+	var it;
+	var r;
+	var i;
+
+	opts = {
+		'iter': 10
+	};
+	rand = randu( opts );
+	it = iterWhileEach( rand, predicate, assert );
+	t.equal( it.next.length, 0, 'has zero arity' );
+
+	expected = [];
+	i = 0;
+	do {
+		r = it.next();
+		if ( r.value ) {
```

Comment:
```suggestion
		if ( typeof r.value !== 'undefined' ) {
```

`r.value` can be `0`.